pkg/services/ngalert/notifier/multiorg_alertmanager.go

```diff
@@ -78,49 +82,119 @@ func (moa *MultiOrgAlertmanager) LoadAndSyncAlertmanagersForOrgs(ctx context.Con
 	return nil
 }
 
+//getLatestConfigs retrieves the latest AlertManager configuration for every organization and returns them as a map where the key is ID of an organization
+func (moa *MultiOrgAlertmanager) getLatestConfigs() (map[int64]*models.AlertConfiguration, error) {
+	configs, err := moa.configStore.GetAllLatestAlertmanagerConfiguration()
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[int64]*models.AlertConfiguration, len(configs))
+	for _, config := range configs {
+		result[config.OrgID] = config
+	}
+
+	return result, nil
+}
+
 func (moa *MultiOrgAlertmanager) SyncAlertmanagersForOrgs(orgIDs []int64) {
 	orgsFound := make(map[int64]struct{}, len(orgIDs))
 	moa.alertmanagersMtx.Lock()
+
+	moa.addOrUpdateAlertManagerConfigurations(orgIDs, orgsFound, func(orgID int64) (*Alertmanager, error) {
```

Comment:
I think it would be more idiomatic Go if this returned `map[int64]struct{}` as I don't think its that clear from the signature that we should be passing an initialized map with no values expecting it to be updated in place? @gotjosh?

Reply:
I had use for it in my follow-up PR but yes, I agree that it would be better to return map